byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +63,139 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
+/** Get the new list of stops from datastore onload */
+function getStops(){
+  const stopList = document.getElementById('stop-list');
+
+  if(stopList != null){
+    stopList.innerText = ""; // clear list
+  }
+
+  fetch('/api/stop')
+  .then(response => response.json())
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      stops.push(stop);
+      var btn = document.createElement('button');
+      btn.id = stop.id;
+      btn.innerText = stop.placename;
+      btn.setAttribute("class", "btn rec-btn");
+      btn.addEventListener("click", function() {
+        deleteFromStops(stop); 
+      });
+     stopList.appendChild(btn);
+    })
+  })
+}
+
+/** Get the new list of stops locally */
+function getStopsList(){
+  const stopList = document.getElementById('stop-list');
+
+  if(stopList != null){
+    stopList.innerText = ""; 
+  }
+
+  // re-render list synchronously
+  stops.forEach((stop)=>{
+    var btn = document.createElement('button');
+    btn.innerText = stop.placename
+    btn.setAttribute("class", "btn rec-btn");
+    btn.addEventListener("click", function() {
+      deleteFromStops(stop); 
+    });
+    stopList.appendChild(btn);
```

Comment:
Can decompose this into a separate function to use across `getStops()` and `getStopsList()`

Reply:
Ah yes, there's only like one line of difference. Will do, thanks for pointing that out!